src/components/memory/ShortTermMemory.tsx

```
@@ -15,6 +29,41 @@ export const ShortTermMemory: React.FC = () => {
   const [searchTerm, setSearchTerm] = useState('');
   const [sortBy, setSortBy] = useState('recent');
   const { shortTermNotes, isLoading, moveToLongTerm, deleteNote, refetch } = useMemoryNotes();
+  const { user } = useAuth();
+  const [showAiInsights, setShowAiInsights] = useState(false);
+
+  const [aiInsights, setAiInsights] = useState<{ summary: string | null; recommendations: any[] }>({ summary: null, recommendations: [] });
+  const [aiInsightsLoading, setAiInsightsLoading] = useState(false);
+
+  // Daily Caching and AI Insights Logic
+  useEffect(() => {
+    if (!user || (user as any).subscription_plan !== 'pro' || !shortTermNotes || shortTermNotes.length === 0) {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)